src/wasm/tree_parse.cpp

```diff
@@ -0,0 +1,93 @@
+#include <iostream>
+#include <fstream>
+#include <nlohmann/json.hpp>
+#include <vector>
+#include <unordered_map>
+#include "tree_parse.hpp"
+
+using json = nlohmann::json;
+
+
+TreeNode parse_json_to_tree(const json& j_node) {
+    TreeNode node;
+    node.name = j_node.value("name", "");
+    node.value = 0;
+    
+    if (j_node.contains("value")) {
+        if (j_node["value"].is_number_unsigned()) {
+            node.value = j_node["value"].get<uint64_t>();
+        } else if (j_node["value"].is_number_integer()) {
+            node.value = static_cast<uint64_t>(j_node["value"].get<int64_t>());
+        }
+    }
+
+    node.cold = j_node.value("cold", false);
+    node.left_sum = 0;
+
+    if (j_node.contains("children")) {
+        for (const auto& child : j_node["children"]) {
+            node.children.push_back(parse_json_to_tree(child));
+        }
+    }
+    return node;
+}
+
+uint64_t calculate_left_sum(TreeNode& node, uint64_t& running_sum) {
+    if (node.children.empty()) {
+        node.left_sum = running_sum; //left_sum is the starting_time for the current function
+        running_sum += node.value;  // sum of leaf values (walltimes) to the left = start_time current
+        } else {
+        for (auto& child : node.children) {
+            calculate_left_sum(child, running_sum);
+        }
+        if (!node.children.empty()) {
+            node.left_sum = node.children.front().left_sum; //parent start_time = left most child start_time
+        }
+    }
+    return node.left_sum;
+}
+
+TreeNode prune_tree(const TreeNode& node, uint64_t threshold_left, uint64_t threshold_right) {
+    TreeNode pruned_node = node;
+
+    pruned_node.children.clear();
+    for (const auto& child : node.children) {
+        //this includes all the functions that have any overlap with the time between the two thresholds
+        if ((child.left_sum + child.value) >= threshold_left && (child.left_sum) <= threshold_right) { 
+            pruned_node.children.push_back(prune_tree(child, threshold_left, threshold_right));
+        }
+    }
+    return pruned_node;
+}
```

Comment:
Good thinking, but your function needs also to slice node values, I'll explain it to you in person :)

Reply:
Done, I only changed for the walltime